Cesium3DTilesSelection/src/CreditSystem.cpp

```diff
@@ -54,6 +56,22 @@ void CreditSystem::addCreditToFrame(Credit credit) {
 void CreditSystem::startNextFrame() noexcept {
   _creditsToNoLongerShowThisFrame.swap(_creditsToShowThisFrame);
   _creditsToShowThisFrame.clear();
+  _creditCounts.clear();
   _currentFrameNumber++;
 }
+
+const std::vector<Credit>& CreditSystem::getCreditsToShowThisFrame() noexcept {
+  // sort credits based on the number of occurrences
+  if (_creditsToShowThisFrame.size() < 2) {
+    return _creditsToShowThisFrame;
+  }
+  const auto& counts = _creditCounts;
+  std::sort(
```

Comment:
I think it's really important to use a stable sort here. Otherwise, credits with equal counts can jump around from frame to frame.

Unfortunately, reading the docs just now, I learned that std::stable_sort will allocate a scratch buffer (presumably on the heap) that is the same size as the container we're sorting. That's really not great for an operation like this that we do every frame.

So I think a good solution is to stick with std::sort and incorporate stability into the comparison function. Something like this (not tested):

```
[&counts](const Credit& a, const Credit& b) {
  int32_t aCounts = counts.at(a.id);
  int32_t bCounts = counts.at(b.id);
  if (aCounts == bCounts)
    return a.id < b.id;
  else
    return aCounts > bCounts;
}
```